apis/vpcresources/v1beta1/vpcresourcerequest_types.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// VPCResourceRequestSpec defines the desired state of VPCResourceRequest
+type VPCResourceRequestSpec struct {
+	RequestTypes  map[string]ResourceType `json:"requestTypes,omitempty"`
+	RequestTarget Target                  `json:"requestTarget,omitempty"`
+	Message       string                  `json:"message,omitempty"`
+}
+
+// Target represents a request's target
+// Target could be pod, node or other Kinds
+type Target struct {
+	Name  string `json:"name,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+// Type represents a request's type
+// ResourceType could features, such as SGP, custom networking, etc.
```

Comment:
Is it possible to strongly type these, or potentially use Enums? What are all of the options here.

Reply:
We want to make this extensible for future cases, for example interfaces, IP addresses, and probably other networking resources. Making this strongly typed could add some limitations on the future use. @jayanthvn do you have any recommendations on this? Thanks

Reply:
An enum should enable you to explicitly expand this over time. It also has an effect of making the interface self-documenting.